api/src/main/java/io/serverlessworkflow/api/deserializers/DataInputSchemaDeserializer.java

```diff
@@ -46,17 +53,43 @@ public DataInputSchema deserialize(JsonParser jp, DeserializationContext ctxt)
     JsonNode node = jp.getCodec().readTree(jp);
 
     DataInputSchema dataInputSchema = new DataInputSchema();
+    JsonNode schemaDefinition = null;
 
-    if (!node.isObject()) {
-      dataInputSchema.setSchema(node.asText());
-      dataInputSchema.setFailOnValidationErrors(true); // default
-
-      return dataInputSchema;
-    } else {
-      dataInputSchema.setSchema(node.get("schema").asText());
+    if (node.isObject() && node.get("schema").isObject() && !node.get("schema").isEmpty()) {
+      schemaDefinition = node.get("schema");
       dataInputSchema.setFailOnValidationErrors(node.get("failOnValidationErrors").asBoolean());
+    } else {
+      String schemaFileDef = node.isObject() ? node.get("schema").asText() : node.asText();
+      dataInputSchema.setFailOnValidationErrors(true);
+      dataInputSchema.setRefValue(schemaFileDef);
+      String schemaFileSrc = Utils.getResourceFileAsString(schemaFileDef);
```

Comment:
nit : renaming schemaFileSrc to schemaFileContent helps in understanding the code flow